### ISSUE TITLE
checkers: remove unnecessary parentheses

### ISSUE THREAD

**Issue by twpayne** [go-critic/go-critic#836]
Currently given the code:

```go
if (a == 0) || (b == 0) {
    // ...
}
```

`go-critic` does not complain. The parentheses are superfluous and can be removed without changing the meaning of the expression.

It would be nice if `go-critic` suggested that the above be replaced with:

```go
if a == 0 || b == 0 {
    // ...
}
```

**Comment by quasilyte** [comment]
We need to define "superfluous" more formally in order for this to work. :)

**Comment by twpayne** [comment]
I would suggest:

"If removing the parentheses does not change the evaluation order of the expression, then they are superfluous."

**Comment by cristaloleg** [comment]
Sounds as an opinionated checker 😉

**Comment by quasilyte** [comment]
Sometimes they are used for logical grouping.

**Comment by quasilyte** [comment]
I want to note that this is doable with AST-based rules:
https://github.com/quasilyte/go-ruleguard/blob/60a948075b45357c18c7e9e3563ba81835a3e670/analyzer/testdata/src/extra/extra.rules.go#L31-L32

```go
m.Match(`($a) || ($b)`).Report(`rewrite as '$a || $b'`)
m.Match(`($a) && ($b)`).Report(`rewrite as '$a && $b'`)
// etc...
```

**Comment by twpayne** [comment]
Closing due to inactivity.
